changelog/10839.improvement.rst

```diff
@@ -0,0 +1 @@
+Synchronous tests that request asynchronous fixtures will now error, instead of silently accepting an unawaited coroutine object as the fixture value.
```

Comment:
Let's add an example, like the one you posted:

```python
@pytest.fixture
async def fix():
    return 1

def test_fix(fix):
    assert 1 == asyncio.run(fix)
```

Is this a valid workaround for the error (did not test it)?

```python
@pytest.fixture
def fix():
    async def func():
        return 1
    return func

def test_fix(fix):
    assert 1 == asyncio.run(fix)
```


If so, we might want to mention this in the changelog as well in case a test suite is using this approach.

Reply:
I added an example to deprecations.rst and linked to it from the changelog :+1: